Page for PX Blue's Flexibility (#293)
pxblue/doc-it: src/docs/design/project-identity.mdx

<!-- KEYWORDS: design tutorial branding brand identity flexible flex -->

# Project Identity

PX Blue provides a foundation for designing and developing applications in a way that makes them feel like they are a part of the same brand. However, the purpose of PX Blue is not to make every application look the same. There is a balance that you must strike between maintaining consistency between applications and establishing a unique brand for each product. PX Blue provides some of the basic elements and patterns to help you build consistency, leaving you more time to focus on what makes your product unique.

<MaterialDesignDescription
    title={'Design Patterns'}
    url={'/patterns'}
    description={`Follow PX Blue's design pattern guidelines.`}
    avatar={<PxblueSmallAltIcon style={{ width: 48, height: 48 }} />}
    minCardHeight={'unset'}
/>

## PX Blue is Flexible

PX Blue is not prescriptive — just as we build on top of Material Design principles, you too can build on top of PX Blue principles to make your products better for your target audience. There are many gray areas and gaps in PX Blue where you are free to make changes and tweaks.

<ImageGrid
    regularWidth

>     images={[ConsumerIndustrial]}

Reviewer: The left image does not really look pxblue, IMO. I don't know if we want to stretch it that far for the example, but I'll leave it up to you.

Author: It assimilates the consumer-facing project I am recently working on. You may find this image to be very "eric".

I am trying to tone the extended app bar gradient a bit further to include more pxblue blue in the gradient spectrum.

Reviewer: I am not surprised to hear that it's an Eric.

    caption={[
        `An example of two projects, both designed to control devices, but with different targeted audience.`,
        `1: A consumer-faced product. End user might be homeowners, small business owners, etc. The project is designed to delight the users.`,
        `2: Industrial product. End user might be asset managers, facility assets managers, etc. The project is designed to  convey a secure and efficient picture for equipment management.`,
    ]}
/>

> We fully expect you to establish your own local design system for your project that extends PX Blue and adds your own flair / style / identity. Depending on your use case, you may want your design system to be more flashy, visual, relaxed, friendly, etc. Depending on where your application will reside, you may also need to change aspects of the design system related to accessibility (such as larger buttons for touch-screens) or legal-requirements in your target markets.

Reviewer: ```suggestion
We expect that you will establish your own local design system for your project that extends PX Blue and adds your own style / identity. Depending on your use case, your design system may need to be more flashy, visual, relaxed, friendly, etc. Depending on where your application will be used, you may also need to change aspects of the design system related to accessibility (such as larger buttons for touch-screens) or legal requirements in your target markets.
```

<ImageGrid
    regularWidth
    images={[StyleChoice]}
    caption={[
        `You may tune styles for your application away from the PX Blue baseline style, as part of your local design system. In this example, the app bar (#1), the input form field (#2) and the buttons (#3) are changed.`,
    ]}
/>

> While establishing your local product identity, remember that PX Blue is here to help establish consistency, so you do not want to go so far down the road of customization that this is lost. Below, we go into details about some common areas of customization as well as some areas that should be avoided.

Reviewer: ```suggestion
While establishing your local product identity, remember that PX Blue is here to help establish consistency. You do not want customize so much that this is lost. Below, we go into details about some common areas of customization as well as some areas that should be avoided.
```

## Common Things to Change

There are some areas where we strongly encourage you to make changes and establish your own brand identity.

Keep in mind that once you've established a local design system, you must make sure that you apply it consistently everywhere. You should use it in your packaging, your website, your email templates, the application, etc.

### Theme

PX Blue [offers two themes](/style/themes): Blue theme and Blue Dark theme. Ideally, an application should support both themes, but some applications might be more usable depending on the environment and context of the usage.

> By default, we use `#007bc1` as our primary theme color. But this is the baseline color and may vary from projects to projects. You may change the primary theme color, given that you carefully design it to avoid inflicting with the color convention ([see below](#color-convention)).

Reviewer: ```suggestion
By default, we use `#007bc1` as our primary theme color, but this can vary from project to project. You may change the primary theme color, given that you carefully design it to avoid conflicting with the color convention ([see below](#color-convention)).
```

    ]}
/>

In certain situations, it might even be beneficial to explore beyond what PX Blue has to offer, _as long as you have a strong reason for it_ (not just "it looks cooler this way").

## Common Branding Efforts

A few UI elements are commonly used to express a product's brand identity, while still remain consistent with the rest of the PX Blue applications.

Note that, once you've developed a local design system, you need to make sure that your brand identity is consistent. That is, they should be under the same design system for packaging, for your website, for the email formats, and for the digital application itself.

### Theme

PX Blue [offers two themes](/style/themes): Blue theme and Blue Dark theme. Ideally, an application should support both themes, but some applications might be more usable depending on the environment and context of the usage.

By default, we use `#007bc1` as our primary theme color. But this is the baseline color and may vary from projects to projects. You may change the primary theme color, given that you carefully design it to avoid inflicting with the color convention ([see below](#color-convention)).

> 

Reviewer: I still feel the same way about changing themes as more of a "shouldn't do unless there is a very good reason". Changing the primary theme color is a quick way to make a product not look like PX Blue anymore.

> 

Reviewer: If we want to just comment out the part about themes until we decide what we want to do about it, I'm fine with that as well.

### Illustrations

<ImageGrid
    images={[IllustrationConsumer, IllustrationIndustrial]}
    caption={[
        `A consumer-oriented product (#1) might use an illustration for empty states, while an industrial product (#2) might use an icon instead.`,
    ]}
/>

[Illustrations](/style/illustrations) are common for [websites](/design/websites) and consumer-faced products. Different image styles can convey different product styles (friendly vs. professional).

### Images

Similiar to illustrations, images are another way to make your applications and websites more consumer-friendly.

**Insert images where we apply background + background images**:

-   App bar
-   Drawer header
-   Drawer header but with user avatar
-   Scorecard header
-   Login

### Icons

We use Material Design's baseline style for [our iconography](/style/iconography).

> However, we understand that many projects have their unique situation and the current iconography set may not cover every single use cases; you are welcome to invent your own icons, or harness [Material Design's community icons](https://materialdesignicons.com/). However, you should make sure that you have thoroughly scoured through Material Design and PX Blue's existing iconographies first, and that you are strictly following [Material Design's icon design guidelines](https://material.io/design/iconography/system-icons.html#design-principles).

Reviewer: ```suggestion
However, we understand that the current iconography set may not cover every single use case or requirement for your product. In these situations, you are welcome to invent your own icons or leverage [Material Design's community icons](https://materialdesignicons.com/). Before going this route, make sure that you have thoroughly explored the available icons first and when creating new icons, ensure that they follow the [Material Design icon design guidelines](https://material.io/design/iconography/system-icons.html#design-principles).
```

### Component Variants

**Insert an image showing variations of buttons and text field**

> Many components have styling variations, and it is up to the designer to decide which one suits the best for the design context. Some common variations include:

Reviewer: ```suggestion
Many components have style variations. These variations are all available for you to use, and it is up to you to decide which ones work best for your application. Some common variations include:
```

> -   Outlined vs. contained (or filled)
> -   Flat vs. elevated
> -   Squared vs. semi-rounded corner vs. rounded corner

Reviewer: ```suggestion
-   Outlined vs. Contained / Filled
-   Flat vs. Elevated
-   Square vs. Rounded Corner vs. Round
```

### Verbiage

**Insert a side-by-side image, one saying "Create a new location" and another saying "Excellent! Now let's create a new location for your business."**

> Your tone can change based on your application nature too. For websites, you want to be appealing and persvasive. For applications, you want to be concise and accurate.
> 

Reviewer: ```suggestion
The actual text used in your application can change based on your intended tone. For websites, you may want to use more persuasive and appealing language to encourage the audience to buy or use your product. Inside an application, you may want to use a more serious tone, especially for applications that deal with potential safety concerns.

```

<MaterialDesignDescription
    title={'Writing'}
    url={`https://material.io/design/communication/writing.html`}

>     description={`Material's principles in writing for user interface.`}

Reviewer: ```suggestion
    description={`Material's language principles for user interface.`}
```

/>

> ## Should NEVER Change

Reviewer: ```suggestion
## Things You Should NEVER Change
```

> Despite all the design freedom you have, there are a few things you should never change. Those either has to do with long-established traditions, or has to do with how PX Blue is established as a branded design system.

Reviewer: ```suggestion
When it comes to PX Blue, you have a great deal of design freedom, but there are a few things you should never change. Most of these are related to maintaining visual consistency and other conventions between applications.
```

> ### Color Convention

Reviewer: ```suggestion
### Status Colors
```

**Insert an image with color misuse on status colors**

> PX Blue has [a fixed set of status color palette](/style/color). The status colors are used to indicate the status of certain items. Do not let colors from other parts of your user interface to interfere with them and give your reader wromg impressions on what is being given with what status.
> 
> In the rare case, there are certain color conventions in the specific industry you are designing for, like green = off, red = open in case of a breaker. Follow the indutrial convention there.

Reviewer: ```suggestion
PX Blue has strict guidelines around our [status color palette](/style/color). The status colors are used to indicate the status of certain items or devices. These should not be changed for any reason — users should be able to instantly know what these colors mean in any PX Blue application. Further, the colors you are using to brand your application should not interfere with these colors. 

The only situation where these may be changed is if there is an industry-specific convention that differs from the PX Blue recommendations (e.g., green = off, red = open for breakers). In these situations, you should follow the industry-specific convention.
```

### Font Family

> PX Blue [uses Open Sans and Noto Sans](/style/typography) for all its applications. We [do encourage you to change font weight and font size](/style/typography#customization) sporatically to display a better information hierarchy, but we strongly discourage you using fonts other than Open Sans and Noto Sans, especially any [Serif fonts](https://en.wikipedia.org/wiki/Serif).

Reviewer: ```suggestion
PX Blue [uses Open Sans and Noto Sans](/style/typography) for all applications. You are encouraged to adjust font [weight and size](/style/typography#customization) where necessary to establish a visual type hierarchy, but we strongly discourage the use of any fonts other than Open Sans and Noto Sans, particularly any [Serif fonts](https://en.wikipedia.org/wiki/Serif).
```

### Consistency

> You shall make sure your local design system is being carried through consistently. Once you pick a style, you should stick to it, instead of changing it every page.

Reviewer: This is pretty much already talked about above

> ### Design Conventions
> 
> Something are just design conventions. Everyone does that.
> 
> Material Design and PX Blue highlight a few things in red. Those are our "do not". You should never include any design patterns like that in your designs.
> 

Reviewer: ```suggestion
### Design Patterns

Throughout our [design patterns](/patterns), we provide examples of "things not to do" — these are typically identified by an image with a red underline. You should never use any of these expressly-forbidden patterns in your products. In all cases, we provide an example of the preferred approach.

```

<!-- KEYWORDS: design tutorial branding brand identity flexible flex -->

# Project Identity and Design Flexibility

PX Blue provides a foundation for designing and developing applications in a way that makes them feel like they are a part of the same brand. However, the purpose of PX Blue is not to make every application look the same.

> As PX Blue serves as a design system for a wide range of applications, some projects might demand unique design or branding elements to better suit their audience. There is a balance that you must strike between maintaining consistency between applications and establishing a unique brand for each product. PX Blue provides some of the basic elements and patterns to help you build consistency, leaving you more time to focus on what makes your product unique.

Reviewer: ```suggestion
As PX Blue serves as a design system for a wide range of applications, some projects might demand unique design or branding elements to better suit their audience. There is a balance that you must strike between maintaining consistency among applications and establishing a unique brand for each product. PX Blue provides some of the basic elements and patterns to help you build consistency, leaving you more time to focus on what makes your product unique.
```

## PX Blue is Flexible

> PX Blue is not prescriptive — using PX Blue to be more efficient in practice shouldn’t mean restricting creativity in a project, or feeling limited on what might be improved.

Reviewer: ```suggestion
PX Blue is not prescriptive — it should make you more efficient without restricting your creativity in a project.
```

When it comes to PX Blue, you have a great deal of design freedom, but there are a few things you should never change. Most of these are related to maintaining visual consistency and other conventions between applications.

> ### Theme
> 
> PX Blue [offers two themes](/style/themes): Blue theme and Blue Dark theme. Ideally, an application should support both themes, but some applications might be more usable depending on the environment and context of the usage.
> 
> You are expected to stay close to those two themes. Both themes have been thoroughly researched to ensure they meet the standards for usability and accessibility.
> 

Reviewer: ```suggestion
### Theme

PX Blue provides [two themes](/style/themes), one light and one dark. Ideally, an application should support both themes, but some situations may call for favoring one over the other (e.g., low-light environments).

You should stay within the bounds of those two themes. Both themes have been thoroughly researched to ensure they meet the standards for usability and accessibility.

> One notable exception to this rule is if you are building products that will be white-labeled or sold under a different brand. In this case, you may find it beneficial to create your own custom themes that will better align with the brand.
```
